frontend/src/app/shared/services/charts.service.ts

```diff
@@ -72,21 +72,25 @@ export class ChartsService {
     let re = new RegExp(query, 'i');
     return this.getCharts(repo).map(charts => {
       return charts.filter(chart => {
-        return chart.attributes.name.match(re) ||
-         chart.attributes.description.match(re) ||
-         chart.attributes.repo.name.match(re) ||
+        return this.checkDefinedMatch(chart.attributes.name, re) ||
+         this.checkDefinedMatch(chart.attributes.description, re) ||
+         this.checkDefinedMatch(chart.attributes.repo.name, re) ||
          this.arrayMatch(chart.attributes.keywords, re) ||
          this.arrayMatch((chart.attributes.maintainers || []).map((m)=> { return m.name }), re) ||
          this.arrayMatch(chart.attributes.sources, re)
       })
     })
   }
 
+  checkDefinedMatch(member, matcher) {
+    return member && member.match(matcher);
+  }
+
   arrayMatch(keywords: string[], re): boolean {
     if(!keywords) return false
 
     return keywords.some((keyword) => {
-      return !!keyword.match(re)
+      return this.checkDefinedMatch(!!keyword, re)
```

Comment:
This was sending `true` to `checkDefinedMatch` and `true.match` is undefined. We actually want to send the string.

```suggestion
      return this.checkDefinedMatch(keyword, re)
```

Reply:
Ah good call… fixed!